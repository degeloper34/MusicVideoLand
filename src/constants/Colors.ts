export default {
  black: "#000",
  white: "#fff",
  gray: "#8f8f8f",
};
