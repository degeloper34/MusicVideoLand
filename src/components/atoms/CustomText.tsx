import Colors from "../../constants/Colors";
import {Text, TextProps} from "react-native";

interface IProps extends TextProps {
  text: string | number;
  type: "regular" | "medium" | "bold";
  fontSize?: number;
  textColor?: string;
  underline?: boolean;
  numberOfLines?: number;
}

export function CustomText({
  type = "medium",
  fontSize = 12,
  underline = false,
  textColor = Colors.white,
  style,
  text,
  numberOfLines,
}: IProps) {
  const decideFontFamilyByType = () => {
    switch (type) {
      case "regular":
        return "sora-regular";
      case "medium":
        return "sora-medium";
      case "bold":
        return "sora-bold";
      default:
        return "sora-regular";
    }
  };

  return (
    <Text
      allowFontScaling={false}
      numberOfLines={numberOfLines}
      style={[
        style,
        {
          fontFamily: decideFontFamilyByType(),
          fontSize: fontSize,
          color: textColor,
          textDecorationLine: underline ? "underline" : "none",
        },
      ]}
    >
      {text}
    </Text>
  );
}
