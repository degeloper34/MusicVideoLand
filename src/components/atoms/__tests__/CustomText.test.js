import renderer from "react-test-renderer";
import React from "react";
import {CustomText} from "../CustomText";
import Colors from "../../../constants/Colors";

describe("CustomText", () => {
  it("renders correctly with text", () => {
    const component = renderer
      .create(<CustomText text={"Hello, degeloper"} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it("renders correctly with text and regular-type", () => {
    const component = renderer
      .create(<CustomText text={"Hello, degeloper"} type={"regular"} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it("renders correctly with text and medium-type", () => {
    const component = renderer
      .create(<CustomText text={"Hello, degeloper"} type={"medium"} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it("renders correctly with text and bold-type", () => {
    const component = renderer
      .create(<CustomText text={"Hello, degeloper"} type={"bold"} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it("renders correctly with white text", () => {
    const component = renderer.create(
      <CustomText
        text={"Hello, degeloper"}
        type={"bold"}
        textColor={Colors.white}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
