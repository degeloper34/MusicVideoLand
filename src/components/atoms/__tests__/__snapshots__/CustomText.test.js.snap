// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomText renders correctly with text 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      undefined,
      Object {
        "color": "#fff",
        "fontFamily": "sora-medium",
        "fontSize": 12,
        "textDecorationLine": "none",
      },
    ]
  }
>
  Hello, degeloper
</Text>
`;

exports[`CustomText renders correctly with text and bold-type 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      undefined,
      Object {
        "color": "#fff",
        "fontFamily": "sora-bold",
        "fontSize": 12,
        "textDecorationLine": "none",
      },
    ]
  }
>
  Hello, degeloper
</Text>
`;

exports[`CustomText renders correctly with text and medium-type 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      undefined,
      Object {
        "color": "#fff",
        "fontFamily": "sora-medium",
        "fontSize": 12,
        "textDecorationLine": "none",
      },
    ]
  }
>
  Hello, degeloper
</Text>
`;

exports[`CustomText renders correctly with text and regular-type 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      undefined,
      Object {
        "color": "#fff",
        "fontFamily": "sora-regular",
        "fontSize": 12,
        "textDecorationLine": "none",
      },
    ]
  }
>
  Hello, degeloper
</Text>
`;

exports[`CustomText renders correctly with white text 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      undefined,
      Object {
        "color": "#fff",
        "fontFamily": "sora-bold",
        "fontSize": 12,
        "textDecorationLine": "none",
      },
    ]
  }
>
  Hello, degeloper
</Text>
`;
